UML editor/src/test/java/UMLMethodsTests.java

```diff
@@ -0,0 +1,191 @@
+        assertFalse(UMLClass.classMap.containsKey("Test"), "Should be prompted the class doesn't exist");
+    }
+
+    /**
+     * Tests removing a non-existing method from an existing class.
+     */
+    @Test
+    public void testRemoveMethodFromNonExistingMethod() {
+        UMLClass.addClass("Test");
+        umlMethods.addMethod("Test", "AddDog", Arrays.asList("int","String"));
+        umlMethods.removeMethod("Test", "RemoveCat");
+
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        assertEquals(1, classInfo.getMethods().size(), "Shouldn't be prompted to enter a valid method");
+    }
+
+    /**
+     * Tests renaming an existing method within a class.
+     */
+    @Test
+    public void testRenameMethod(){
+        UMLClass.addClass("Test");
+        umlMethods.addMethod("Test", "AddDog", Arrays.asList("int","String"));
+        umlMethods.renameMethod("Test", "AddDog", "AddMultipleDogs");
+
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        UMLMethodInfo methodInfo = classInfo.getMethodByName("AddMultipleDogs");
+
+        assertNotNull(methodInfo);
+        assertEquals("AddMultipleDogs", methodInfo.getMethodName());
+    }
+
+    /**
+     * Tests renaming a non-existing method within a class.
+     */
+    @Test
+    public void testRenameMethodFromNonExistingMethod() {
+        UMLClass.addClass("Test");
+        umlMethods.renameMethod("Test", "AddDog", "AddMultipleDogs");
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        assertNull(classInfo.getMethodByName("AddMultipleDogs"));
+    }
+
+    /**
+     * Tests renaming a method to an already existing method's name within the same class.
+     */
+    @Test
+    public void testRenameMethodToExistingName(){
+        UMLClass.addClass("Test");
+        umlMethods.addMethod("Test", "AddDog", Arrays.asList("int","String"));
+        umlMethods.addMethod("Test", "RemoveDog", Arrays.asList("float"));
+        umlMethods.renameMethod("Test", "AddDog", "RemoveDog");
+
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        // Should not change either name
+        assertNotNull(classInfo.getMethodByName("AddDog"));
+        assertNotNull(classInfo.getMethodByName("RemoveDog"));
+        assertEquals(2, classInfo.getMethods().size());
+    }
+
+    /**
+     * Tests adding a parameter to an existing method within a class.
+     */
+    @Test
+    public void testAddParm(){
+        UMLClass.addClass("Test");
+        umlMethods.addMethod("Test", "AddDog", Arrays.asList("int"));
+
+        umlMethods.addParameter("Test", "AddDog", "String");
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        UMLMethodInfo methodInfo = classInfo.getMethodByName("AddDog");
+        assertTrue(methodInfo.getParameters().contains("String"));
+    }
+
+    /**
+     * Tests removing a parameter from an existing method within a class.
+     */
+    @Test
```

Comment:
Small typo where int is added to the parameter list but String is expected and "removed", test passes since String was never in the array to begin with.

Needs fixed

Reply:
Thank you for the review, I have fixed that typo.